static-assets/components/cstudio-view-controllers/history.js

```diff
@@ -462,33 +465,44 @@
-                  revertActionEl.innerHTML =
-                    '<span id="actionRevert' + version.versionNumber + '" class="action fa fa-reply"></span>';
-                  revertActionEl.item = selection;
-                  revertActionEl.version = version.versionNumber;
-                  new YAHOO.widget.Tooltip('tooltipRevert' + revertActionEl.version, {
-                    context: 'actionRevert' + revertActionEl.version,
+                  $revertDropdown = $(
+                    `<div class="${dropdownClass} inline-block relative confirmation-dropdown">
+                        <span id="actionRevert${
+                      version.versionNumber
+                    }" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="action fa fa-reply"></span>
+                        <ul class="dropdown-menu pull-right" aria-labelledby="actionRevert${version.versionNumber}">
+                          <li><a class="cancel" href="#" onclick="return false;">${formatMessage(words.cancel)}</a></li>
+                          <li role="separator" class="divider"></li>
+                          <li><a class="confirm" href="#">${formatMessage(words.confirm)}</a></li>
+                        </ul>
+                      </div>`
+                  ).appendTo($(col5El));
+                  $revertDropdown.data('item', selection);
+                  $revertDropdown.data('version', version.versionNumber);
+
+                  new YAHOO.widget.Tooltip('tooltipRevert' + version.versionNumber, {
+                    context: 'actionRevert' + version.versionNumber,
                     container: _this.tooltipsContainer,
                     text: CMgs.format(formsLangBundle, 'historyDialogRevertFileMessage'),
                     zIndex: 104103
                   });
-
-                  col5El.appendChild(revertActionEl);
                 }
 
                 (function(item) {
                   if (_this.isWrite) {
-                    Event.addListener(revertActionEl, 'click', function() {
-                      CStudioAuthoring.Service.revertContentItem(CStudioAuthoringContext.site, item, this.version, {
+                    $revertDropdown.find('.confirm').on('click', function(e) {
+                      e.preventDefault();
+                      const $dropdown = $(this).closest('.confirmation-dropdown');
+
+                      CStudioAuthoring.Service.revertContentItem(CStudioAuthoringContext.site, item, $dropdown.data('version'), {
                         success: function() {
                           if (CStudioAuthoringContext.isPreview) {
                             CStudioAuthoring.Operations.refreshPreview();
                           }
                           eventNS.data = item;
                           document.dispatchEvent(eventNS);
 
-                          _this.loadHistory(_this.selection);
+                          _this.loadConfigurationHistory(_this.selection);
```

Comment:
Looks like this will use config history for all files; not just config files. Does this work for non config files?

Reply:
Yes if you see the code, there are two diff functions, loadConfigurationHistory, and loadHistory, config files are using loadConfigurationHistory, and the other are using loadHistory